diff --git a/dist/index.js b/dist/index.js
index 857bd0b9d652110b434a8b696260ec23b18c5f69..7d7bb0a367c8d7e155de8fbf6fce7fc72ef6e050 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1968,6 +1968,11 @@ function atEndOfCell(view, axis, dir) {
   if (!(view.state.selection instanceof TextSelection3))
     return null;
   const { $head } = view.state.selection;
+  if ($head.parent.type.spec.tableRole == "cell" || $head.parent.type.spec.tableRole == "header_cell") {
+    const cellPos = $head.before();
+    const dirStr = axis == "vert" ? dir > 0 ? "down" : "up" : dir > 0 ? "right" : "left";
+    return view.endOfTextblock(dirStr) ? cellPos : null;
+  }
   for (let d = $head.depth - 1; d >= 0; d--) {
     const parent = $head.node(d), index = dir < 0 ? $head.index(d) : $head.indexAfter(d);
     if (index != (dir < 0 ? 0 : parent.childCount))
